internal/storage/mount.go

```diff
@@ -67,3 +75,41 @@ func UnmountPath(ctx context.Context, target string, removeTarget bool) (err err
 	}
 	return nil
 }
+
+func UnmountAllInPath(ctx context.Context, path string, removeTarget bool) (err error) {
+	childMounts, err := listMountPointsUnderPath(path)
+	if err != nil {
+		return err
+	}
+
+	for i := len(childMounts) - 1; i >= 0; i-- {
+		childPath := childMounts[i]
+		if err := UnmountPath(ctx, childPath, removeTarget); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func ListMountPointsUnderPath(path string) ([]string, error) {
+	var mountPoints []string
+	f, err := os.Open(procMountFile)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		line := scanner.Text()
+		fields := strings.Split(line, " ")
+		if len(fields) < numProcMountFields {
+			continue
+		}
+		destPath := fields[1]
+		if strings.HasPrefix(destPath, path) {
+			mountPoints = append(mountPoints, destPath)
+		}
+	}
+
+	return mountPoints, nil
```

Comment:
I think you will need to apply some sort order. Is there some guarantee that `/proc/self/mountinfo` is ordered in a way that we will always create the array in an unmount order?

Reply:
I believe /proc/mounts should be listed in the order that the mounts were issued.

Reply:
Ok as long as that is true then this will always work in reverse order.